File: FindSmallestLetterGreaterThanTarget.cpp
class Solution {
public:
    char nextGreatestLetter(vector<char>& letters, char target) {
        
        
        int len = letters.size();
        int l = 0, h = len - 1;
        
        char ans = letters[0];
        
        while(l <= h){
            int m = (l + h)/2;
            if(letters[m] > target){
                ans = letters[m];
                h = m - 1;
            }
            else{
                l = m + 1;
            }
        }
        
        return ans;
    }
};

File: FindSmallestLetterGreaterThanTarget.java
class Solution {
    public char nextGreatestLetter(char[] letters, char target) {
        
        int n = letters.length;
        int l = 0, h = n - 1;
        while(l < h){
            int m = (h - l)/2 + l;
            if(letters[m] <= target){
                l = m+ 1; 
            }else h = m;
        }

        return letters[h] <= target ? letters[0] : letters[h];
    }
}
